### PR TITLE
Update CHANGELOG for 0.11.0 release

### PR DESCRIPTION
**Pull request by amartinezfayo**
Update CHANGELOG for 0.11.0 release.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,21 @@
 # Changelog
 
+## [0.11.0] - TBD
+
+### What's New
+- Introduced refactored server APIs (#1533, #1548, #1563, #1567, #1568, #1571, #1575, #1576, #1577, #1578, #1582, #1585, #1586, #1587, #1588, #1589, #1590, #1591, #1592, #1593, #1594, #1595, #1597, #1604, #1606, #1607, #1613, #1615, #1617, #1622, #1623, #1628, #1630, #1633, #1641, #1643, #1646, #1647, #1654, #1659, #1667, #1673, #1674, #1683, #1684, #1689, #1690, #1692, #1693, #1694, #1701, #1708, #1727, #1728, #1730, #1733, #1734, #1739, #1749, #1753, #1768, #1772, #1779)
+- Unix workloads can now be attested by user groups (#1771)
```
Comment (azdagron) [inline CHANGELOG.md:7]:
```suggestion
- Unix workloads can now be attested using auxilliary group membership (#1771)
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,21 @@
...
+
+### What's Changed
+- Federation is now a stable feature (#1656, #1737, #1777)
+- Removed support for the `UpstreamCA` plugin, which was deprecated in favor of the `UpstreamAuthority` plugin in v0.10.0 (#1699)
+- Removed deprecated `upstream_bundle` server configurable. The server now always use the upstream bundle as the trust bundle (#1702)
+- The server's AWS node attestor subsumed all the functionality of the node resolver, which has been deprecated (#1705)
+- Removed pluggability of the DataStore interface, restricting use to the current built-in `sql` plugin (#1707)
+- Unknown config options now make the server and agent fail to start (#1714)
+- Added registration entry revision number functionality (#1720)
```
Comment (azdagron) [inline CHANGELOG.md:16]:
```suggestion
- Improved registration entry change detection on agent (#1720)
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,21 @@
...
+- Fixed to have `/tmp/agent.sock` as the default socket path in the agent when it is not configured (#1738)
```
Comment (azdagron) [inline CHANGELOG.md:17]:
```suggestion
- `/tmp/agent.sock` is now the default socket path for the agent (#1738)
```

File: CHANGELOG.md (hunk 1 above)
Comment (amartinezfayo) [reply CHANGELOG.md:7]: Is there a typo here where it should be `auxiliary` vs `auxilliary`?

Comment (azdagron) [reply CHANGELOG.md:7]: There sure is!

**Pull request merged**